statistikk/src/main/kotlin/no/nav/su/se/bakover/statistikk/behandling/søknadsbehandling/BehandlingsstatistikkMapperForSøknadsbehandling.kt

```diff
@@ -0,0 +1,145 @@
+            resultatBegrunnelse = utledAvslagsgrunner(this.søknadsbehandling.avslagsgrunner),
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Underkjent.Innvilget -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.søknadsbehandling.attesteringer.toFunksjonellTid(this.søknadsbehandling.id, clock),
+            behandlingStatus = BehandlingStatus.Underkjent,
+            behandlingsresultat = BehandlingResultat.Innvilget,
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Underkjent.Avslag -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.søknadsbehandling.attesteringer.toFunksjonellTid(this.søknadsbehandling.id, clock),
+            behandlingStatus = BehandlingStatus.Underkjent,
+            behandlingsresultat = BehandlingResultat.AvslagSøknadsbehandling,
+            resultatBegrunnelse = utledAvslagsgrunner(this.søknadsbehandling.avslagsgrunner),
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Iverksatt.Innvilget -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.vedtak.opprettet,
+            behandlingStatus = BehandlingStatus.Iverksatt,
+            behandlingsresultat = BehandlingResultat.Innvilget,
+            beslutter = this.vedtak.attestant,
+            avsluttet = true,
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Iverksatt.Avslag -> toDto(
+            clock = clock,
+            gitCommit = gitCommit,
+            funksjonellTid = this.vedtak.opprettet,
+            behandlingStatus = BehandlingStatus.Iverksatt,
+            behandlingsresultat = BehandlingResultat.AvslagSøknadsbehandling,
+            resultatBegrunnelse = utledAvslagsgrunner(this.søknadsbehandling.avslagsgrunner),
+            beslutter = this.vedtak.attestant,
+            avsluttet = true,
+        )
+
+        is StatistikkEvent.Behandling.Søknad.Lukket -> {
+            toDto(
+                clock = clock,
+                gitCommit = gitCommit,
+                funksjonellTid = this.søknadsbehandling.lukketTidspunkt,
+                behandlingStatus = BehandlingStatus.Avsluttet,
+                behandlingsresultat = this.søknadsbehandling.toBehandlingResultat(),
+                // Det kreves ikke attestant ved lukking.
+                beslutter = null,
+                totrinnsbehandling = false,
+                avsluttet = true,
+                saksbehandler = this.saksbehandler,
+            )
+        }
+    }
+}
+
+private fun StatistikkEvent.Behandling.Søknad.toDto(
+    clock: Clock,
+    gitCommit: GitCommit?,
+    funksjonellTid: Tidspunkt,
+    behandlingStatus: BehandlingStatus,
+    behandlingsresultat: BehandlingResultat? = null,
+    resultatBegrunnelse: String? = null,
+    beslutter: NavIdentBruker.Attestant? = null,
+    totrinnsbehandling: Boolean = true,
+    avsluttet: Boolean = false,
```

Comment:
Usikker på om det er bedre eller verre med default verdier her, virker som det er veldig mange ting man må/bør ta stilling til for hvert enkelt tilfelle for at dette skal bli "rett". Default gjør det veldig enkelt å hoppe over å ta stilling til ting.

Reply:
Fjernet defaultverdier.